test.php

```diff
@@ -2,107 +2,116 @@
+            'host' => getenv('DATABASE_URL'),
+            'driver' => 'pdo_pgsql',
+        ]);
 
-   public function getClient(): ?BigQueryClient
-   {
-       if ($this->client) {
-           return $this->client;
-       }
+        return $this->connection;
+    }
 
-       return $this->client = new BigQueryClient([
-           /*Temporary username on bigquery is the project id and host holds the keyFile content*/
-           'projectId' => $this->importerConfig->getUsername(),
-           'keyFile' => json_decode($this->importerConfig->getHost(), true),
-           'scopes' => [BigQueryClient::SCOPE],
-       ]);
-   }
+    /**
+     * @return BigQueryClient|null
+     */
+    public function getClient(): ?BigQueryClient
+    {
+        if (null !== $this->client) {
+            return $this->client;
+        }
 
-   /**
-    * @return ImportedValue[]|array
-    *
-    * @throws Exception
-    */
-   protected function executeQuery(QueryBuilder $query, FieldConfiguration $fieldConfiguration): ?array
-   {
-       $importValues = [];
-       $parser = new StdMathParser();
-       /** @var FunctionNode $AST */
-       $AST = $parser->parse($fieldConfiguration->getCalculationString());
-       $evaluator = new Evaluator();
-       $client = $this->getClient();
-       $queryJobConfig = $client->query($query->getSQL(), [
-           'useQueryCache' => true,
-       ]);
-       $queryResults = $client->runQuery($queryJobConfig);
-       $results = $queryResults->rows();
-       $count = 0;
-       foreach ($results as $result) {
-           ++$count;
-           $variables = [];
-           foreach ($fieldConfiguration->getSourceFields() as $sourceField) {
-               $variables[$sourceField->getLabel()] = $result[$sourceField->getLabel()];
-           }
-           $evaluator->setVariables($variables);
-           $importValue = new ImportedValue();
-           $value = null;
-           try {
-               $value = $AST->accept($evaluator);
-           } catch (DivisionByZeroException $e) {
-               //TODO Log this maybe
-           }
-           if (null !== $value && 0.0 != $value) {
-               $importValue->setValue($value)
-                   ->setTimestamp(new DateTime($result[$this->dateField]))
-                   ->setAsin($result[$this->asinField]);
-               $importValues[] = $importValue;
-           }
-       }
+        return $this->client = new BigQueryClient([
+            'projectId' => $this->importerConfig->getUsername(),
+            'keyFile' => json_decode($this->importerConfig->getHost(), true),
+            'scopes' => [BigQueryClient::SCOPE],
+        ]);
+    }
 
-       return $importValues;
-   }
+    /**
+     * @return ImportedValue[]|array
+     */
+    protected function executeQuery(QueryBuilder $query, FieldConfiguration $fieldConfiguration): array
+    {
+        $results = $this->getQueryResults($query);
+
+        return $this->getImportedValues($fieldConfiguration, $results);
```

Comment:
I split this function into two more to separate the complexity and see it more readable.